File: java-advent-of-code/src/test/java/advent/twenty_sixteen/Day07InternetProtocolVersionSevenTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day07InternetProtocolVersionSevenTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day07InternetProtocolVersionSeven("/twenty_sixteen/Day07-InternetProtocolVersionSeven-input.txt");

        assertEquals(110, dp.getPart1Answer().intValue());
        assertEquals(242, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day07InternetProtocolVersionSeven("/twenty_sixteen/Day07-InternetProtocolVersionSeven-example#1.txt");

        assertEquals(2, dp.getPart1Answer().intValue());
        assertEquals(0, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day07InternetProtocolVersionSeven("/twenty_sixteen/Day07-InternetProtocolVersionSeven-example#2.txt");

        assertEquals(0, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day07InternetProtocolVersionSevenTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day07InternetProtocolVersionSevenTest extends FlatSpec with Matchers {
  "2016 Day 7 - Input" should "calculate answers" in {
    val dp = new Day07InternetProtocolVersionSeven("/twenty_sixteen/Day07-InternetProtocolVersionSeven-input.txt")

    dp.part1Answer should be(110)
    dp.part2Answer should be(242)
  }

  "2016 Day 7 - Example #1" should "calculate answers" in {
    val dp = new Day07InternetProtocolVersionSeven("/twenty_sixteen/Day07-InternetProtocolVersionSeven-example#1.txt")

    dp.part1Answer should be(2)
    dp.part2Answer should be(0)
  }

  "2016 Day 7 - Example #2" should "calculate answers" in {
    val dp = new Day07InternetProtocolVersionSeven("/twenty_sixteen/Day07-InternetProtocolVersionSeven-example#2.txt")

    dp.part1Answer should be(0)
    dp.part2Answer should be(3)
  }
}